hoodie-client/src/main/java/com/uber/hoodie/table/HoodieCopyOnWriteTable.java

```diff
@@ -231,7 +230,7 @@ protected HoodieMergeHandle getUpdateHandle(String commitTime, String fileId,
   }
 
   protected HoodieMergeHandle getUpdateHandle(String commitTime, String fileId,
-      Map<String, HoodieRecord<T>> keyToNewRecords, Optional<HoodieDataFile> dataFileToBeMerged) {
+      Map<String, HoodieRecord<T>> keyToNewRecords, HoodieDataFile dataFileToBeMerged) {
```

Comment:
Around assuming there is always a base file for updates. How would small insert handling in MOR work? Do we always write a parquet file if one does not exist in a ingest batch today? I don't know if that's the model we want to emulate in future though.. Should we able to first write data onto logs even without parquet files and later batch compact them together ? otherwise, this will block streaming ingest rght